catalyst/callbacks/aggregation.py

```diff
@@ -0,0 +1,139 @@
+        prefix: str,
+        metrics: Union[str, List[str], Dict[str, float]] = None,
+        mode: str = "mean",
+        scope: str = "batch",
+        multiplier: float = 1.0,
+    ) -> None:
+        """
+        Args:
+            prefix: new key for aggregated metric.
+            metrics (Union[str, List[str], Dict[str, float]]): If not None,
+                it aggregates only the values from the metric by these keys.
+                for ``weighted_sum`` aggregation it must be a Dict[str, float].
+            mode: function for aggregation.
+                Must be either ``sum``, ``mean`` or ``weighted_sum`` or user's
+                function to aggregate metrics. This function must get dict of
+                metrics and runner and return aggregated metric. It can be
+                useful for complicated fine tuning with different losses that
+                depends on epochs and loader or something also
+            scope: type of metric. Must be either ``batch``, ``loader`` or
+                ``epoch``
+            multiplier: scale factor for the aggregated metric.
+        Examples:
+            Loss depends on epoch(Note that all loss functions that are used
+            in the aggregation function must be defined.)
+            >>> from catalyst.dl import MetricAggregationCallback
+            >>>
+            >>> def aggregation_function(metrics, runner):
+            >>>     epoch = runner.stage_epoch_step
+            >>>     region_loss = metrics["loss_dice"] + metrics["loss_iou"]
+            >>>     bce_loss = metrics['loss_bce']
+            >>>     loss = 1 / epoch * bce_loss + epoch / 3 * region_loss
+            >>>     return loss
+            >>>
+            >>> MetricAggregationCallback(
+            >>>     mode=aggregation_function,
+            >>>     prefix='loss',
+            >>> )
+        """
+        super().__init__(
+            order=CallbackOrder.metric_aggregation, node=CallbackNode.all
+        )
+
+        if prefix is None or not isinstance(prefix, str):
+            raise ValueError("prefix must be str")
+
+        if mode in ("sum", "mean"):
+            if metrics is not None and not isinstance(metrics, list):
+                raise ValueError(
+                    "For `sum` or `mean` mode the metrics must be "
+                    "None or list or str (not dict)"
+                )
+        elif mode in ("weighted_sum", "weighted_mean"):
+            if metrics is None or not isinstance(metrics, dict):
+                raise ValueError(
+                    "For `weighted_sum` or `weighted_mean` mode "
+                    "the metrics must be specified "
+                    "and must be a dict"
+                )
+        elif not callable(mode):
+            raise NotImplementedError(
+                "mode must be `sum`, `mean` "
+                "or `weighted_sum` or `weighted_mean` or be Callable"
+            )
+
+        assert scope in ("batch", "loader", "epoch")
```

Comment:
there is no `epoch` case handler, should we remove it?

Reply:
It seems to us that we do not need it. Since epoch_metrics is just copied from loader_metrics

Reply:
let's remove than